src/50_Samples_%26_Templates/Product.html

```diff
@@ -198,6 +245,24 @@
     <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
     </head>
     <body>
+    <amp-state id="allColors">
```

Comment:
Huh, good question. The `amp-bind` expression language is a bit limited. It would be nice if `Object.keys` were available. Is there any chance of getting that whitelisted? (`Object.values`, too.) Is there any reason why various `String.*` and `Array.*` methods are available, but no `Object.*`?

So, to make sure I understand, you need to determine the first available size so that an available product is selected on page load? I don't completely understand the example nor `amp-bind` but perhaps the easiest way given that it doesn't seem to be possible to loop through the data structure is to define another variable that stores the "on load" state:

```javascript
{
  'size': 'S',
  'color': 'green'
}
```